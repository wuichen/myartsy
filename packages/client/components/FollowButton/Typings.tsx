export interface FollowTrackingData {
  contextModule?: string
  entity_id?: string
  entity_slug?: string
  entity_type?: string
}
