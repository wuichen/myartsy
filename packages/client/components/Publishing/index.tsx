// Components
export { Article } from "./Article"
export { Artwork } from "./Sections/Artwork"
export { Authors } from "./Sections/Authors"
export { Caption } from "./Sections/Caption"
export { Embed } from "./Sections/Embed"
export { FullscreenViewer } from "./Sections/FullscreenViewer/FullscreenViewer"
export { Header } from "./Header/Header"
export { InstantArticleEmailSignup } from "./Email/InstantArticleEmailSignup"
export { Image } from "./Sections/Image"
export { ImageCollection } from "./Sections/ImageCollection"
export { ImageSetPreview } from "./Sections/ImageSetPreview"
export {
  ImageSetPreviewClassic,
} from "./Sections/ImageSetPreview/ImageSetPreviewClassic"
export { Nav } from "./Nav/Nav"
export { SeriesLayout } from "./Layouts/SeriesLayout"
export { Text } from "./Sections/Text"
export { Video } from "./Sections/Video"
export { VideoLayout } from "./Layouts/VideoLayout"
export { VideoCover } from "./Video/VideoCover"
export { VideoAbout } from "./Video/VideoAbout"
export { PartnerInline } from "./Partner/PartnerInline"
export { PartnerBlock } from "./Partner/PartnerBlock"
export {
  RelatedArticlesCanvas,
} from "./RelatedArticles/Canvas/RelatedArticlesCanvas"
export {
  RelatedArticlesPanel,
} from "./RelatedArticles/Panel/RelatedArticlesPanel"

// Icon SVGs
export { IconArtist } from "./Icon/IconArtist"
export { IconBlockquote } from "./Icon/IconBlockquote"
export { IconClearFormatting } from "./Icon/IconClearFormatting"
export { IconDrag } from "./Icon/IconDrag"
export { IconEditEmbed } from "./Icon/IconEditEmbed"
export { IconEditImages } from "./Icon/IconEditImages"
export { IconEditSection } from "./Icon/IconEditSection"
export { IconEditText } from "./Icon/IconEditText"
export { IconEditVideo } from "./Icon/IconEditVideo"
export { IconHamburger } from "./Icon/IconHamburger"
export { IconHeroImage } from "./Icon/IconHeroImage"
export { IconHeroVideo } from "./Icon/IconHeroVideo"
export { IconImageFullscreen } from "./Icon/IconImageFullscreen"
export { IconImageSet } from "./Icon/IconImageSet"
export { IconLayoutFullscreen } from "./Icon/IconLayoutFullscreen"
export { IconLayoutSplit } from "./Icon/IconLayoutSplit"
export { IconLayoutText } from "./Icon/IconLayoutText"
export { IconLayoutBasic } from "./Icon/IconLayoutBasic"
export { IconLink } from "./Icon/IconLink"
export { IconOrderedList } from "./Icon/IconOrderedList"
export { IconVideoPlay } from "./Icon/IconVideoPlay"
export { IconRemove } from "./Icon/IconRemove"
export { IconUnorderedList } from "./Icon/IconUnorderedList"
export { IconLock } from "./Icon/IconLock"

// FIXME: Refactor out SizeMe; see https://github.com/ctrlplusb/react-sizeme#server-side-rendering
import sizeMe from "react-sizeme"
sizeMe.noPlaceholders = true

// Constants
import * as AllConstants from "./Constants"
export const Constants = AllConstants
