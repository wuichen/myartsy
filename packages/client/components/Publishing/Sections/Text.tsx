import ReactHtmlParser, { convertNodeToElement } from "@artsy/react-html-parser"
import { startsWith } from "lodash"
import React, { Component } from "react"
import { get } from "Utils/get"
import { ArticleLayout } from "../Typings"
import { StyledText } from "./StyledText"

interface Props extends React.HTMLProps<HTMLDivElement> {
  color?: string
  html?: string
  isContentEnd?: boolean
  isContentStart?: boolean
  layout: ArticleLayout
  postscript?: boolean
  showTooltips?: boolean
}

interface State {
  html: string
}

export class Text extends Component<Props, State> {
  static defaultProps = {
    color: "black",
    showTooltips: false,
  }

  state = {
    html: this.props.html || "",
  }

  componentDidMount() {
    const html = this.htmlMaybeWithContentEnd()

    this.setState({ html })
  }

  htmlMaybeWithContentEnd = () => {
    const { html } = this.state
    const { isContentEnd } = this.props
    // Remove existing spans - TODO: Backfill out of articles
    const cleanedHtml = html.replace("<span class='content-end'> </span>", "")

    if (isContentEnd) {
      const doc = document.createElement("div")
      doc.innerHTML = cleanedHtml

      const paragraphs = doc.getElementsByTagName("P")
      const lastParagraph =
        paragraphs.length && paragraphs[paragraphs.length - 1]

      if (lastParagraph) {
        // insert content-end in last paragraph
        lastParagraph.innerHTML =
          lastParagraph.innerHTML + "<span class='content-end'> </span>"
      }
      return doc.innerHTML
    }
    return cleanedHtml
  }

  shouldShowTooltipForURL = (node: Element) => {
    const urlBase = "https://www.artsy.net/"
    const types = ["artist/", "gene/"]

    for (const type of types) {
      if (startsWith(node.getAttribute("href"), urlBase + type)) {
        return true
      }
    }

    return false
  }

  transformNode = (node: Element, index) => {
    const { color } = this.props
    // Dont include relay components unless necessary
    // To avoid 'regeneratorRuntime' error
    const LinkWithTooltip = require("../ToolTip/LinkWithTooltip").default

    if (node.tagName === "P") {
      const newNode = node.ownerDocument.createElement("div")
      newNode.setAttribute("class", "paragraph")
      Array.from(node.childNodes).forEach(child => newNode.appendChild(child))
      return convertNodeToElement(newNode, index, this.transformNode)
    }

    if (node.tagName === "A" && this.shouldShowTooltipForURL(node)) {
      const href = node.getAttribute("href")
      const linkNode = node.childNodes[0]

      if (linkNode && linkNode.textContent) {
        const props = { key: href + index, url: href, color }
        const next = node.nextSibling
        const text = linkNode.textContent

        // Check to see if there's an apostrophe following a linked section of
        // text and if found, return it.
        const apostrophe = get(next, n => {
          const str = n.textContent.substr(0, 2)
          if (/[â€™'][a-zA-Z]/.test(str)) {
            return str
          }
        })

        if (apostrophe) {
          // Remove the apostrophe from the original text
          next.textContent = next.textContent.substring(2)
          // And wrap the whole thing with a span preventing whitespace breaks
          return (
            <span
              className="preventLineBreak"
              key={`apostropheLinkNode-${props.key}`}
            >
              <LinkWithTooltip {...props}>{text}</LinkWithTooltip>
              {apostrophe}
            </span>
          )
        }

        return <LinkWithTooltip {...props}>{text}</LinkWithTooltip>
      }
    }
  }

  render() {
    const {
      children,
      color,
      isContentStart,
      layout,
      postscript,
      showTooltips,
    } = this.props
    const { html } = this.state

    return (
      <StyledText
        className="article__text-section"
        color={color}
        isContentStart={isContentStart}
        layout={layout}
        postscript={postscript}
        showTooltips={showTooltips}
      >
        {html.length ? (
          showTooltips ? (
            <div>
              {ReactHtmlParser(html, { transform: this.transformNode })}
            </div>
          ) : (
            <div dangerouslySetInnerHTML={{ __html: html }} />
          )
        ) : (
          children
        )}
      </StyledText>
    )
  }
}
