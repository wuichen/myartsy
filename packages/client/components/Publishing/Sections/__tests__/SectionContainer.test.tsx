import { SectionData, SectionLayout } from "Components/Publishing/Typings"
import "jest-styled-components"
import React from "react"
import renderer from "react-test-renderer"
import {
  getSectionMobilePadding,
  getSectionWidth,
  SectionContainer,
} from "../SectionContainer"
import { Text } from "../Text"

describe("SectionContainer", () => {
  let section: SectionData

  beforeEach(() => {
    section = {
      layout: "overflow_fillwidth" as SectionLayout,
      body: "<p>Hello, world!</p>",
      type: "text",
    }
  })

  describe("#getSectionWidth", () => {
    it("returns column_width by default", () => {
      expect(getSectionWidth()).toBe("680px")
    })

    it("returns a custom width if prop provided", () => {
      expect(getSectionWidth(section, "feature", 900)).toBe("900px")
    })

    describe("classic layout", () => {
      const articleLayout = "classic"

      it("returns correct width for overflow_fillwidth", () => {
        expect(getSectionWidth(section, articleLayout)).toBe("1100px")
      })

      it("returns correct width for column_width", () => {
        section.layout = "column_width"
        expect(getSectionWidth(section, articleLayout)).toBe("580px")
      })
    })

    describe("standard layout", () => {
      const articleLayout = "standard"

      it("returns correct width for overflow_fillwidth", () => {
        expect(getSectionWidth(section, articleLayout)).toBe("780px")
      })

      it("returns correct width for column_width", () => {
        section.layout = "column_width"
        expect(getSectionWidth(section, articleLayout)).toBe("680px")
      })

      it("returns correct width for blockquotes", () => {
        section.body = "<blockquote>Hello, world!</blockquote>"
        expect(getSectionWidth(section, articleLayout)).toBe("780px")
      })
    })

    describe("feature layout", () => {
      const articleLayout = "feature"

      it("returns correct width for overflow_fillwidth", () => {
        expect(getSectionWidth(section, articleLayout)).toBe("780px")
      })

      it("returns correct width for column_width", () => {
        section.layout = "column_width"
        expect(getSectionWidth(section, articleLayout)).toBe("680px")
      })

      it("returns correct width for fillwidth", () => {
        section.layout = "fillwidth"
        expect(getSectionWidth(section, articleLayout)).toBe("100%")
      })

      it("returns correct width for blockquotes", () => {
        section.body = "<blockquote>Hello, world!</blockquote>"
        expect(getSectionWidth(section, articleLayout)).toBe("900px")
      })
    })
  })

  describe("#getSectionMobilePadding", () => {
    it("returns with padding by default", () => {
      expect(getSectionMobilePadding()).toBe("0 20px")
    })

    it("returns correct padding for video", () => {
      section.type = "video"
      expect(getSectionMobilePadding(section)).toBe(0)
    })

    it("returns correct padding for image_collection", () => {
      section.type = "image_collection"
      expect(getSectionMobilePadding(section)).toBe(0)
    })

    it("returns correct padding for image_set", () => {
      section.type = "image_set"
      expect(getSectionMobilePadding(section)).toBe(0)
    })

    it("returns correct padding for mini image_set", () => {
      section.type = "image_set"
      section.layout = "mini"
      expect(getSectionMobilePadding(section)).toBe("0 20px")
    })
  })

  describe("Snapshots", () => {
    it("renders overflow_fillwidth properly", () => {
      const sectionContainer = renderer
        .create(
          <SectionContainer section={section} articleLayout="standard">
            <Text html={section.body} layout="standard" />
          </SectionContainer>
        )
        .toJSON()
      expect(sectionContainer).toMatchSnapshot()
    })

    it("renders fillwidth properly", () => {
      section.layout = "fillwidth"
      const sectionContainer = renderer
        .create(
          <SectionContainer section={section} articleLayout="standard">
            <Text html={section.body} layout="standard" />
          </SectionContainer>
        )
        .toJSON()
      expect(sectionContainer).toMatchSnapshot()
    })

    it("renders column_width properly", () => {
      section.layout = "column_width"
      const sectionContainer = renderer
        .create(
          <SectionContainer section={section} articleLayout="standard">
            <Text html={section.body} layout="standard" />
          </SectionContainer>
        )
        .toJSON()
      expect(sectionContainer).toMatchSnapshot()
    })
  })
})
