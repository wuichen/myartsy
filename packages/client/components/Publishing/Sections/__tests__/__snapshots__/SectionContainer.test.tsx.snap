// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionContainer Snapshots renders column_width properly 1`] = `
.c0 {
  box-sizing: border-box;
  margin: auto;
  margin-bottom: 40px;
  width: 680px;
  max-width: 100%;
}

.c1 {
  position: relative;
  width: 100%;
  color: black;
}

.c1 a {
  color: black;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-image: linear-gradient( to bottom,transparent 0,#333 1px,transparent 0 );
  background-size: 1.25px 4px;
  background-repeat: repeat-x;
  background-position: bottom;
}

.c1 a:hover {
  color: #C2C2C2;
  opacity: 1;
}

.c1 div[class*='ToolTip'] a {
  background-image: none;
  opacity: 1;
  color: inherit;
}

.c1 p,
.c1 ul,
.c1 ol,
.c1 .paragraph,
.c1 div[data-block=true] .public-DraftStyleDefault-block {
  font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
  font-size: 23px;
  line-height: 1.5em;
  -webkit-font-smoothing: antialiased;
  padding-top: 1em;
  padding-bottom: 1em;
  margin: 0;
  font-style: inherit;
}

.c1 p:first-child,
.c1 .paragraph:first-child,
.c1 div[data-block=true]:first-child .public-DraftStyleDefault-block {
  padding-top: 0;
}

.c1 p:last-child,
.c1 .paragraph:last-child,
.c1 div[data-block=true]:last-child .public-DraftStyleDefault-block {
  padding-bottom: 0;
}

.c1 ul,
.c1 ol {
  padding-left: 1em;
}

.c1 ul {
  list-style: disc;
}

.c1 ol {
  list-style: decimal;
}

.c1 li {
  font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
  font-size: 23px;
  line-height: 1.5em;
  -webkit-font-smoothing: antialiased;
  padding-top: .5em;
  padding-bottom: .5em;
}

.c1 h1 {
  font-family: Unica77LLWebRegular,Arial,serif;
  -webkit-font-smoothing: antialiased;
  font-size: 40px;
  line-height: 1.1em;
  font-weight: normal;
  padding-top: 107px;
  padding-bottom: 46px;
  margin: 0;
  position: relative;
  text-align: center;
}

.c1 h1::before {
  content: "";
  width: 8px;
  height: 8px;
  background: black;
  border-radius: 50%;
  position: absolute;
  top: 69px;
  right: calc(50% - 4px);
}

.c1 h2 {
  font-family: Unica77LLWebRegular,Arial,serif;
  -webkit-font-smoothing: antialiased;
  font-size: 32px;
  line-height: 1.1em;
  font-weight: normal;
  margin: 0;
}

.c1 h2 a {
  background-size: 1.25px 1px;
}

.c1 h2 .c2 {
  background-position: bottom !important;
}

.c1 h3 {
  font-family: Unica77LLWebRegular,Arial,serif;
  -webkit-font-smoothing: antialiased;
  font-size: 19px;
  line-height: 1.5em;
  font-weight: normal;
  padding-top: 23px;
  margin: 0;
}

.c1 h3 strong {
  font-weight: normal;
  font-family: Unica77LLWebMedium,Arial,serif;
  -webkit-font-smoothing: antialiased;
  font-size: 19px;
  line-height: 1.5em;
}

.c1 h3 a {
  background-size: 1.25px 1px;
}

.c1 blockquote {
  font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
  font-size: 40px;
  line-height: 1.1em;
  -webkit-font-smoothing: antialiased;
  text-align: left;
  font-weight: normal;
  padding-top: 46px;
  padding-bottom: 46px;
  margin: 0;
  word-break: break-word;
}

.c1 .preventLineBreak {
  white-space: nowrap;
}

.c1 .content-end {
  display: inline-block;
  content: "";
  width: 8px;
  height: 8px;
  background: black;
  border-radius: 50%;
  margin-left: 12px;
  margin-bottom: 1px;
}

.c1 .artist-follow {
  vertical-align: middle;
  margin-left: 10px;
  cursor: pointer;
  background: none transparent;
}

.c1 .artist-follow::before {
  font-family: "artsy-icons";
  content: "";
  vertical-align: -8.5px;
  line-height: 32px;
  font-size: 32px;
}

.c1 .artist-follow::after {
  content: "Follow";
  font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
  font-size: 17px;
  line-height: 1.1em;
  -webkit-font-smoothing: antialiased;
  text-transform: none;
}

@media (max-width:1280px) {
  .c0 {
    width: 680px;
  }
}

@media (max-width:900px) {
  .c0 {
    padding: 0 20px;
  }
}

@media (max-width:600px) {
  .c1 p,
  .c1 ul,
  .c1 ol,
  .c1 div[data-block=true] .public-DraftStyleDefault-block {
    font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
    font-size: 19px;
    line-height: 1.5em;
    -webkit-font-smoothing: antialiased;
  }

  .c1 li {
    font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
    font-size: 19px;
    line-height: 1.5em;
    -webkit-font-smoothing: antialiased;
  }

  .c1 h1 {
    font-family: Unica77LLWebRegular,Arial,serif;
    -webkit-font-smoothing: antialiased;
    font-size: 34px;
    line-height: 1.1em;
  }

  .c1 h2 {
    font-family: Unica77LLWebRegular,Arial,serif;
    -webkit-font-smoothing: antialiased;
    font-size: 32px;
    line-height: 1.1em;
  }

  .c1 h3 {
    font-family: Unica77LLWebRegular,Arial,serif;
    -webkit-font-smoothing: antialiased;
    font-size: 16px;
    line-height: 1.1em;
    line-height: 1.5em;
  }

  .c1 h3 strong {
    font-family: Unica77LLWebMedium,Arial,serif;
    -webkit-font-smoothing: antialiased;
    font-size: 16px;
    line-height: 1.1em;
  }

  .c1 blockquote {
    font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
    font-size: 34px;
    line-height: 1.1em;
    -webkit-font-smoothing: antialiased;
    margin: 0;
  }

  .c1 .content-start {
    font-size: 55px;
  }
}

<div
  className="c0"
>
  <div
    className="article__text-section c1"
    color="black"
  >
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p>Hello, world!</p>",
        }
      }
    />
  </div>
</div>
`;

exports[`SectionContainer Snapshots renders fillwidth properly 1`] = `
.c0 {
  box-sizing: border-box;
  margin: auto;
  margin-bottom: 40px;
  width: 680px;
  max-width: 100%;
}

.c1 {
  position: relative;
  width: 100%;
  color: black;
}

.c1 a {
  color: black;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-image: linear-gradient( to bottom,transparent 0,#333 1px,transparent 0 );
  background-size: 1.25px 4px;
  background-repeat: repeat-x;
  background-position: bottom;
}

.c1 a:hover {
  color: #C2C2C2;
  opacity: 1;
}

.c1 div[class*='ToolTip'] a {
  background-image: none;
  opacity: 1;
  color: inherit;
}

.c1 p,
.c1 ul,
.c1 ol,
.c1 .paragraph,
.c1 div[data-block=true] .public-DraftStyleDefault-block {
  font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
  font-size: 23px;
  line-height: 1.5em;
  -webkit-font-smoothing: antialiased;
  padding-top: 1em;
  padding-bottom: 1em;
  margin: 0;
  font-style: inherit;
}

.c1 p:first-child,
.c1 .paragraph:first-child,
.c1 div[data-block=true]:first-child .public-DraftStyleDefault-block {
  padding-top: 0;
}

.c1 p:last-child,
.c1 .paragraph:last-child,
.c1 div[data-block=true]:last-child .public-DraftStyleDefault-block {
  padding-bottom: 0;
}

.c1 ul,
.c1 ol {
  padding-left: 1em;
}

.c1 ul {
  list-style: disc;
}

.c1 ol {
  list-style: decimal;
}

.c1 li {
  font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
  font-size: 23px;
  line-height: 1.5em;
  -webkit-font-smoothing: antialiased;
  padding-top: .5em;
  padding-bottom: .5em;
}

.c1 h1 {
  font-family: Unica77LLWebRegular,Arial,serif;
  -webkit-font-smoothing: antialiased;
  font-size: 40px;
  line-height: 1.1em;
  font-weight: normal;
  padding-top: 107px;
  padding-bottom: 46px;
  margin: 0;
  position: relative;
  text-align: center;
}

.c1 h1::before {
  content: "";
  width: 8px;
  height: 8px;
  background: black;
  border-radius: 50%;
  position: absolute;
  top: 69px;
  right: calc(50% - 4px);
}

.c1 h2 {
  font-family: Unica77LLWebRegular,Arial,serif;
  -webkit-font-smoothing: antialiased;
  font-size: 32px;
  line-height: 1.1em;
  font-weight: normal;
  margin: 0;
}

.c1 h2 a {
  background-size: 1.25px 1px;
}

.c1 h2 .c2 {
  background-position: bottom !important;
}

.c1 h3 {
  font-family: Unica77LLWebRegular,Arial,serif;
  -webkit-font-smoothing: antialiased;
  font-size: 19px;
  line-height: 1.5em;
  font-weight: normal;
  padding-top: 23px;
  margin: 0;
}

.c1 h3 strong {
  font-weight: normal;
  font-family: Unica77LLWebMedium,Arial,serif;
  -webkit-font-smoothing: antialiased;
  font-size: 19px;
  line-height: 1.5em;
}

.c1 h3 a {
  background-size: 1.25px 1px;
}

.c1 blockquote {
  font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
  font-size: 40px;
  line-height: 1.1em;
  -webkit-font-smoothing: antialiased;
  text-align: left;
  font-weight: normal;
  padding-top: 46px;
  padding-bottom: 46px;
  margin: 0;
  word-break: break-word;
}

.c1 .preventLineBreak {
  white-space: nowrap;
}

.c1 .content-end {
  display: inline-block;
  content: "";
  width: 8px;
  height: 8px;
  background: black;
  border-radius: 50%;
  margin-left: 12px;
  margin-bottom: 1px;
}

.c1 .artist-follow {
  vertical-align: middle;
  margin-left: 10px;
  cursor: pointer;
  background: none transparent;
}

.c1 .artist-follow::before {
  font-family: "artsy-icons";
  content: "";
  vertical-align: -8.5px;
  line-height: 32px;
  font-size: 32px;
}

.c1 .artist-follow::after {
  content: "Follow";
  font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
  font-size: 17px;
  line-height: 1.1em;
  -webkit-font-smoothing: antialiased;
  text-transform: none;
}

@media (max-width:1280px) {
  .c0 {
    width: 680px;
  }
}

@media (max-width:900px) {
  .c0 {
    padding: 0 20px;
  }
}

@media (max-width:600px) {
  .c1 p,
  .c1 ul,
  .c1 ol,
  .c1 div[data-block=true] .public-DraftStyleDefault-block {
    font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
    font-size: 19px;
    line-height: 1.5em;
    -webkit-font-smoothing: antialiased;
  }

  .c1 li {
    font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
    font-size: 19px;
    line-height: 1.5em;
    -webkit-font-smoothing: antialiased;
  }

  .c1 h1 {
    font-family: Unica77LLWebRegular,Arial,serif;
    -webkit-font-smoothing: antialiased;
    font-size: 34px;
    line-height: 1.1em;
  }

  .c1 h2 {
    font-family: Unica77LLWebRegular,Arial,serif;
    -webkit-font-smoothing: antialiased;
    font-size: 32px;
    line-height: 1.1em;
  }

  .c1 h3 {
    font-family: Unica77LLWebRegular,Arial,serif;
    -webkit-font-smoothing: antialiased;
    font-size: 16px;
    line-height: 1.1em;
    line-height: 1.5em;
  }

  .c1 h3 strong {
    font-family: Unica77LLWebMedium,Arial,serif;
    -webkit-font-smoothing: antialiased;
    font-size: 16px;
    line-height: 1.1em;
  }

  .c1 blockquote {
    font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
    font-size: 34px;
    line-height: 1.1em;
    -webkit-font-smoothing: antialiased;
    margin: 0;
  }

  .c1 .content-start {
    font-size: 55px;
  }
}

<div
  className="c0"
>
  <div
    className="article__text-section c1"
    color="black"
  >
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p>Hello, world!</p>",
        }
      }
    />
  </div>
</div>
`;

exports[`SectionContainer Snapshots renders overflow_fillwidth properly 1`] = `
.c0 {
  box-sizing: border-box;
  margin: auto;
  margin-bottom: 40px;
  width: 780px;
  max-width: 100%;
}

.c1 {
  position: relative;
  width: 100%;
  color: black;
}

.c1 a {
  color: black;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-image: linear-gradient( to bottom,transparent 0,#333 1px,transparent 0 );
  background-size: 1.25px 4px;
  background-repeat: repeat-x;
  background-position: bottom;
}

.c1 a:hover {
  color: #C2C2C2;
  opacity: 1;
}

.c1 div[class*='ToolTip'] a {
  background-image: none;
  opacity: 1;
  color: inherit;
}

.c1 p,
.c1 ul,
.c1 ol,
.c1 .paragraph,
.c1 div[data-block=true] .public-DraftStyleDefault-block {
  font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
  font-size: 23px;
  line-height: 1.5em;
  -webkit-font-smoothing: antialiased;
  padding-top: 1em;
  padding-bottom: 1em;
  margin: 0;
  font-style: inherit;
}

.c1 p:first-child,
.c1 .paragraph:first-child,
.c1 div[data-block=true]:first-child .public-DraftStyleDefault-block {
  padding-top: 0;
}

.c1 p:last-child,
.c1 .paragraph:last-child,
.c1 div[data-block=true]:last-child .public-DraftStyleDefault-block {
  padding-bottom: 0;
}

.c1 ul,
.c1 ol {
  padding-left: 1em;
}

.c1 ul {
  list-style: disc;
}

.c1 ol {
  list-style: decimal;
}

.c1 li {
  font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
  font-size: 23px;
  line-height: 1.5em;
  -webkit-font-smoothing: antialiased;
  padding-top: .5em;
  padding-bottom: .5em;
}

.c1 h1 {
  font-family: Unica77LLWebRegular,Arial,serif;
  -webkit-font-smoothing: antialiased;
  font-size: 40px;
  line-height: 1.1em;
  font-weight: normal;
  padding-top: 107px;
  padding-bottom: 46px;
  margin: 0;
  position: relative;
  text-align: center;
}

.c1 h1::before {
  content: "";
  width: 8px;
  height: 8px;
  background: black;
  border-radius: 50%;
  position: absolute;
  top: 69px;
  right: calc(50% - 4px);
}

.c1 h2 {
  font-family: Unica77LLWebRegular,Arial,serif;
  -webkit-font-smoothing: antialiased;
  font-size: 32px;
  line-height: 1.1em;
  font-weight: normal;
  margin: 0;
}

.c1 h2 a {
  background-size: 1.25px 1px;
}

.c1 h2 .c2 {
  background-position: bottom !important;
}

.c1 h3 {
  font-family: Unica77LLWebRegular,Arial,serif;
  -webkit-font-smoothing: antialiased;
  font-size: 19px;
  line-height: 1.5em;
  font-weight: normal;
  padding-top: 23px;
  margin: 0;
}

.c1 h3 strong {
  font-weight: normal;
  font-family: Unica77LLWebMedium,Arial,serif;
  -webkit-font-smoothing: antialiased;
  font-size: 19px;
  line-height: 1.5em;
}

.c1 h3 a {
  background-size: 1.25px 1px;
}

.c1 blockquote {
  font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
  font-size: 40px;
  line-height: 1.1em;
  -webkit-font-smoothing: antialiased;
  text-align: left;
  font-weight: normal;
  padding-top: 46px;
  padding-bottom: 46px;
  margin: 0;
  word-break: break-word;
}

.c1 .preventLineBreak {
  white-space: nowrap;
}

.c1 .content-end {
  display: inline-block;
  content: "";
  width: 8px;
  height: 8px;
  background: black;
  border-radius: 50%;
  margin-left: 12px;
  margin-bottom: 1px;
}

.c1 .artist-follow {
  vertical-align: middle;
  margin-left: 10px;
  cursor: pointer;
  background: none transparent;
}

.c1 .artist-follow::before {
  font-family: "artsy-icons";
  content: "";
  vertical-align: -8.5px;
  line-height: 32px;
  font-size: 32px;
}

.c1 .artist-follow::after {
  content: "Follow";
  font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
  font-size: 17px;
  line-height: 1.1em;
  -webkit-font-smoothing: antialiased;
  text-transform: none;
}

@media (max-width:1280px) {
  .c0 {
    width: 680px;
  }
}

@media (max-width:900px) {
  .c0 {
    padding: 0 20px;
  }
}

@media (max-width:600px) {
  .c1 p,
  .c1 ul,
  .c1 ol,
  .c1 div[data-block=true] .public-DraftStyleDefault-block {
    font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
    font-size: 19px;
    line-height: 1.5em;
    -webkit-font-smoothing: antialiased;
  }

  .c1 li {
    font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
    font-size: 19px;
    line-height: 1.5em;
    -webkit-font-smoothing: antialiased;
  }

  .c1 h1 {
    font-family: Unica77LLWebRegular,Arial,serif;
    -webkit-font-smoothing: antialiased;
    font-size: 34px;
    line-height: 1.1em;
  }

  .c1 h2 {
    font-family: Unica77LLWebRegular,Arial,serif;
    -webkit-font-smoothing: antialiased;
    font-size: 32px;
    line-height: 1.1em;
  }

  .c1 h3 {
    font-family: Unica77LLWebRegular,Arial,serif;
    -webkit-font-smoothing: antialiased;
    font-size: 16px;
    line-height: 1.1em;
    line-height: 1.5em;
  }

  .c1 h3 strong {
    font-family: Unica77LLWebMedium,Arial,serif;
    -webkit-font-smoothing: antialiased;
    font-size: 16px;
    line-height: 1.1em;
  }

  .c1 blockquote {
    font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
    font-size: 34px;
    line-height: 1.1em;
    -webkit-font-smoothing: antialiased;
    margin: 0;
  }

  .c1 .content-start {
    font-size: 55px;
  }
}

<div
  className="c0"
>
  <div
    className="article__text-section c1"
    color="black"
  >
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p>Hello, world!</p>",
        }
      }
    />
  </div>
</div>
`;
